nexus/src/app/sagas/region_replacement_drive.rs

```diff
@@ -465,17 +465,69 @@ async fn check_from_previous_propolis_step(
         return Ok(DriveCheck::ActionRequired);
     };
 
+    // `migration_id` is set at the beginning of an instance migration (before
+    // anything has happened), and is cleared at the end (after the migration is
+    // finished but before the migration target activates disk Volumes). For
+    // now, return `DriveCheck::Wait`, and pick up driving the region
+    // replacement forward after the migration has completed.
+    //
+    // If this saga does not wait, it will interleave with the instance
+    // migration saga. Depending on Nexus' view of what stage the migration is
+    // in, volume replacement requests could be sent to the source propolis or
+    // destination propolis.
```

Comment:
To clarify: any individual call to `instance_fetch_with_vmm` will always produce a VMM that is either a migration source or not migrating. But if this function's caller captured an ongoing migration from A to B, and a later operation recaptures the instance/VMM, it may observe that the `active_propolis_id` is now B (because migration might have finished).

Reply:
:+1: this is required clarification, I dropped it in 0bd3f355e